nixos/modules/programs/preload.nix

```diff
@@ -0,0 +1,172 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.programs.preload;
+  settingsFormat = pkgs.formats.ini {};
+  configFile = settingsFormat.generate "preload.conf" cfg.settings;
+in {
+  options = {
+    programs.preload = {
+      enable = mkEnableOption "Preload loads files and executables to memory to speed up processes";
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = settingsFormat.type;
+        };
+
+        default = {
+          model = {
+            cycle = 20;
+            usecorrelation = "true";
+            minsize = 2000000;
+            memtotal = -10;
+            memfree = 50;
+            memcached = 0;
+          };
+          system = {
+            doscan = "true";
+            dopredict = "true";
+            autosave = 3600;
+            mapprefix = "/run/current-system/sw/share;/run/current-system/sw/lib;!/";
+            exeprefix = "!/run/current-system/sw/sbin/;/run/current-system/sw/bin;!/";
```

Comment:
@GustavoPeredo are you sure it works fine? the things that i am most concerned about are these 2